test/QuestFactory.spec.ts

```diff
@@ -416,5 +416,35 @@ describe('QuestFactory', () => {
         deployedFactoryContract.mintReceipt(erc20QuestId, messageHash, signature)
       ).to.be.revertedWithCustomError(questFactoryContract, 'QuestEnded')
     })
+
+    it('should revert if the mint fee is insufficient', async function () {
+      const requiredFee = 1000
+      await erc20Quest.queue()
+      await deployedFactoryContract.setMintFee(requiredFee)
+      await time.setNextBlockTimestamp(startDate)
+
+      await expect(
+        deployedFactoryContract.mintReceipt(erc20QuestId, messageHash, signature, {
+          value: requiredFee - 1,
+        })
+      ).to.be.revertedWith('Insufficient mint fee')
+    })
+
+    it('should succeed if the mint fee is equal or greator than required amount, and only recieve the required amount', async function () {
+      const requiredFee = 1000
+      await erc20Quest.queue()
+      await deployedFactoryContract.setMintFee(requiredFee)
+      await time.setNextBlockTimestamp(startDate)
+      const balanceBefore = await ethers.provider.getBalance(deployedFactoryContract.getMintFeeRecipient())
+
+      await expect(
+        deployedFactoryContract.mintReceipt(erc20QuestId, messageHash, signature, {
+          value: requiredFee + 100,
```

Comment:
Since we're sending more than `requiredFee` can we also have a test to check that the refund excess works?